src/main/app/cypress/integration/frontPage.test.ts

```diff
@@ -72,6 +72,19 @@ describe('frontPage', () => {
     );
   });
 
+  it('adds organization to favourites', () => {
+    cy.get('#organization-favourites').should('not.exist');
+    cy.get('.favourite-icon__organization').first().click();
+    cy.get('#organization-favourites').should('exist');
+  });
+
+  it('removes organization from favourites', () => {
+    cy.get('#organization-favourites').within(() => {
+      cy.get('.favourite-icon__organization').first().click();
+    });
+    cy.get('#organization-favourites').should('not.exist');
+  });
+
```

Comment:
UI-testeissä ei tule käyttää id:itä ja luokkia, koska käyttäjä ei näe niitä. Jos on pakko käyttää, jotain id:tä, niin sitten `data-cy`-attribuuttia ja testing-libraryn `cy.findByTestId()`-funktiota. Tätä pitäisi kuitenkin käytännössä tarvita hyvin harvoin. Jos oikeaa elementtiä ei saa haettua muilla testing-libraryn funktioilla (esim. cy.findByTole tai cy.findByLabelText`), se yleensä tarkoittaa sitä, että elementti ei ole saavutettava, ja asia tulisi korjata. https://testing-library.com/docs/cypress-testing-library/intro

Reply:
Tästä olen törmännyt toisenlaisiinkin näkemyksiin, ja riippuu toki mitä testillä halutaan todentaa. Tekstillä haku on usein aika lailla hitaampi (riippuu toki käytetystä testityökalusta, enemmän kokemusta robot frameworkista) ja joskus on tehtävä myös kompromisseja sen suhteen kauanko testiajojen on järkevää kestää. Ihan tyypillinen patterni testiautomaatiokehittäjilläkin tuntuu olevan test-id:n käyttö. 
Täsäs myös ylläpidettävyydelle seurauksia: jos fiksataan automaattitestit tekstisisältöön, tarkoittaa helposti sitä että virkaihmisen tekemä tekstitysmuutos tuottaakin buildien failaamisen, ja monesti sillä että tekstit on ylläpidettävissä koodareista riippumattomuutta tavoitellaan juuri sitä että tekstityksiä voisi päivittää ilman uuden version asenteluja.

Reply:
mutta joo, kiinnitin kans huomiota tuohon että se classilla haku ei ehkä ole se tyypillisin tapa, mutta en nyt tullut kumminkaan kommentoineeksi kun kokemus tuosta reframesta on vielä ohkaisempi ja ajattelin että jos tuo esim on siinä kiertotapa siksi että on hankalaa lisäillä niitä testId:itä tms, kun sitä nyt näemmä on käytetty

Reply:
Ajattelisin näin, että se millä napataan kiinni tarkasteltavaksi haluttu dom-elementti on yksi asia ja assertiot että siinä on käyttäjänäkökulmasta ne tekstit ja attribuutit mitä tarvitsee on asia jonka ei tarvitse olla sidonnainen siihen millä se elementti on haettu. Jos haluaa automaatiolla varmistaa saavutettavuutta, ne attribuutit on sitten tsekattava vilä erikseen kun esim ruudunlukijatekstit eivät välttämättä ole yksi yhteen niiden kälitekstien kanssa (näitä tikettejähän nyt on ollut että kälissä näkyvä asia ei tule luetuksi ruudunlukijalla).